### PR TITLE
kafka src/binding type bumps

### PR DESCRIPTION
**Pull request by matzew**
Signed-off-by: Matthias Wessendorf <mwessend@redhat.com>

Fixes #

## Proposed Changes

- bump types to `v1beta1` 

**Release Note**

```release-note
NONE (missing reconcilers, etc. just the types).
```

**Docs**

### CODE REVIEW

File: kafka/source/pkg/apis/sources/v1beta1/doc.go [hunk 1]
```diff
@@ -0,0 +1,22 @@
+/*
+Copyright 2019 The Knative Authors
```
Comment (mattmoor) [inline kafka/source/pkg/apis/sources/v1beta1/doc.go:2]: Incorrect copyright year:
```suggestion
Copyright 2020 The Knative Authors
```

File: kafka/source/pkg/apis/sources/v1beta1/kafka_lifecycle.go [hunk 2]
```diff
@@ -0,0 +1,110 @@
+/*
+Copyright 2019 The Knative Authors
```
Comment (mattmoor) [inline kafka/source/pkg/apis/sources/v1beta1/kafka_lifecycle.go:2]: Incorrect copyright year:
```suggestion
Copyright 2020 The Knative Authors
```

File: kafka/source/pkg/apis/sources/v1beta1/kafka_lifecycle_test.go [hunk 3]
```diff
@@ -0,0 +1,220 @@
+/*
+Copyright 2019 The Knative Authors
```
Comment (mattmoor) [inline kafka/source/pkg/apis/sources/v1beta1/kafka_lifecycle_test.go:2]: Incorrect copyright year:
```suggestion
Copyright 2020 The Knative Authors
```

File: kafka/source/pkg/apis/sources/v1beta1/kafka_types.go [hunk 4]
```diff
@@ -0,0 +1,128 @@
+/*
+Copyright 2019 The Knative Authors
```
Comment (mattmoor) [inline kafka/source/pkg/apis/sources/v1beta1/kafka_types.go:2]: Incorrect copyright year:
```suggestion
Copyright 2020 The Knative Authors
```

File: kafka/source/pkg/apis/sources/v1beta1/kafka_types_test.go [hunk 5]
```diff
@@ -0,0 +1,30 @@
+/*
+Copyright 2019 The Knative Authors
```
Comment (mattmoor) [inline kafka/source/pkg/apis/sources/v1beta1/kafka_types_test.go:2]: Incorrect copyright year:
```suggestion
Copyright 2020 The Knative Authors
```

File: kafka/source/pkg/apis/sources/v1beta1/kafka_validation.go [hunk 6]
```diff
@@ -0,0 +1,46 @@
+/*
+Copyright 2019 The Knative Authors
```
Comment (mattmoor) [inline kafka/source/pkg/apis/sources/v1beta1/kafka_validation.go:2]: Incorrect copyright year:
```suggestion
Copyright 2020 The Knative Authors
```

File: kafka/source/pkg/apis/sources/v1beta1/kafka_validation_test.go [hunk 7]
```diff
@@ -0,0 +1,183 @@
+/*
+Copyright 2019 The Knative Authors
```
Comment (mattmoor) [inline kafka/source/pkg/apis/sources/v1beta1/kafka_validation_test.go:2]: Incorrect copyright year:
```suggestion
Copyright 2020 The Knative Authors
```

File: kafka/source/pkg/apis/sources/v1beta1/register.go [hunk 8]
```diff
@@ -0,0 +1,59 @@
+/*
+Copyright 2019 The Knative Authors
```
Comment (mattmoor) [inline kafka/source/pkg/apis/sources/v1beta1/register.go:2]: Incorrect copyright year:
```suggestion
Copyright 2020 The Knative Authors
```

File: kafka/source/pkg/apis/sources/v1beta1/register_test.go [hunk 9]
```diff
@@ -0,0 +1,69 @@
+/*
+Copyright 2019 The Knative Authors
```
Comment (mattmoor) [inline kafka/source/pkg/apis/sources/v1beta1/register_test.go:2]: Incorrect copyright year:
```suggestion
Copyright 2020 The Knative Authors
```

File: kafka/source/pkg/apis/sources/v1beta1/kafka_types.go [hunk 10]
```diff
@@ -0,0 +1,128 @@
...
+	// +required
+	Topics []string `json:"topics"`
+
+	// ConsumerGroupID is the consumer group ID.
+	// +optional
+	ConsumerGroup string `json:"consumerGroup,omitempty"`
+
+	// Sink is a reference to an object that will resolve to a domain name to use as the sink.
+	// +optional
+	Sink *duckv1.Destination `json:"sink,omitempty"`
```
Comment (n3wscott) [inline kafka/source/pkg/apis/sources/v1beta1/kafka_types.go:81]: Flatten this to import the base source duck:

```
	// inherits duck/v1 SourceSpec, which currently provides:
	// * Sink - a reference to an object that will resolve to a domain name or
	//   a URI directly to use as the sink.
	// * CloudEventOverrides - defines overrides to control the output format
	//   and modifications of the event sent to the sink.
	duckv1.SourceSpec `json:",inline"`
```

File: kafka/source/pkg/apis/sources/v1beta1/kafka_types.go [hunk 11]
```diff
@@ -0,0 +1,128 @@
...
+
+	// ServiceAccoutName is the name of the ServiceAccount that will be used to run the Receive
+	// Adapter Deployment.
+	ServiceAccountName string `json:"serviceAccountName,omitempty"`
```
Comment (n3wscott) [inline kafka/source/pkg/apis/sources/v1beta1/kafka_types.go:85]: is this needed?

File: kafka/source/pkg/apis/sources/v1beta1/kafka_validation_test.go [hunk 12]
```diff
@@ -0,0 +1,194 @@
...
+			},
+			allowed: false,
+		},
+		"Bootstrap servers changed": {
+			orig: &fullSpec,
+			updated: KafkaSourceSpec{
+				KafkaAuthSpec: bindingsv1beta1.KafkaAuthSpec{
+					BootstrapServers: []string{"server1,server2"},
+				},
+				SourceSpec:               fullSpec.SourceSpec,
```
Comment (mattmoor) [inline kafka/source/pkg/apis/sources/v1beta1/kafka_validation_test.go:73]: Format Go code:
```suggestion
				SourceSpec:         fullSpec.SourceSpec,
```

File: kafka/source/pkg/apis/sources/v1alpha1/kafka_conversion.go [hunk 13]
```diff
@@ -0,0 +1,128 @@
...
+			Sink:               source.Spec.Sink.DeepCopy(),
+		}
+		if reflect.DeepEqual(*sink.Spec.Sink, duckv1.Destination{}) {
+			sink.Spec.Sink = nil
+		}
+		return nil
+	default:
+		return fmt.Errorf("Unknown conversion, got: %T", source)
+	}
+}
```
Comment (mattmoor) [inline kafka/source/pkg/apis/sources/v1alpha1/kafka_conversion.go:128]: Add trailing newline:
```suggestion
}

```

File: kafka/source/pkg/apis/sources/v1beta1/kafka_conversion.go [hunk 14]
```diff
@@ -0,0 +1,33 @@
...
+
+// ConvertTo implements apis.Convertible
+func (source *KafkaSource) ConvertTo(ctx context.Context, sink apis.Convertible) error {
+	return fmt.Errorf("v1beta1 is the highest known version, got: %T", sink)
+}
+
+// ConvertFrom implements apis.Convertible
+func (sink *KafkaSource) ConvertFrom(ctx context.Context, source apis.Convertible) error {
+	return fmt.Errorf("v1beta1 is the highest known version, got: %T", source)
+}
```
Comment (mattmoor) [inline kafka/source/pkg/apis/sources/v1beta1/kafka_conversion.go:33]: Add trailing newline:
```suggestion
}

```

File: kafka/source/pkg/apis/sources/v1alpha1/kafka_conversion.go [hunk 15]
```diff
@@ -0,0 +1,128 @@
...
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	"context"
+	"fmt"
```
Comment (mattmoor) [inline kafka/source/pkg/apis/sources/v1alpha1/kafka_conversion.go:21]: Format Go code:
```suggestion
	"fmt"
	"reflect"

```

File: kafka/source/pkg/apis/sources/v1alpha1/kafka_conversion.go [hunk 16]
```diff
@@ -0,0 +1,128 @@
...
+	"knative.dev/eventing-contrib/kafka/source/pkg/apis/bindings/v1alpha1"
+	bindingsv1beta1 "knative.dev/eventing-contrib/kafka/source/pkg/apis/bindings/v1beta1"
+	"knative.dev/eventing-contrib/kafka/source/pkg/apis/sources/v1beta1"
+	"knative.dev/pkg/apis"
+	duckv1 "knative.dev/pkg/apis/duck/v1"
+	"reflect"
```
Comment (mattmoor) [inline kafka/source/pkg/apis/sources/v1alpha1/kafka_conversion.go:27]: Format Go code:
```suggestion

```

File: kafka/source/pkg/apis/sources/v1alpha1/kafka_conversion.go [hunk 17]
```diff
@@ -0,0 +1,128 @@
...
+	switch sink := obj.(type) {
+	case *v1beta1.KafkaSource:
+		sink.ObjectMeta = source.ObjectMeta
+		sink.Spec = v1beta1.KafkaSourceSpec{
+			KafkaAuthSpec: bindingsv1beta1.KafkaAuthSpec{
+				BootstrapServers: source.Spec.KafkaAuthSpec.BootstrapServers,
+				Net:              bindingsv1beta1.KafkaNetSpec{
+					SASL: bindingsv1beta1.KafkaSASLSpec{
+						Enable:   source.Spec.KafkaAuthSpec.Net.SASL.Enable,
+						User:     bindingsv1beta1.SecretValueFromSource{
```
Comment (mattmoor) [inline kafka/source/pkg/apis/sources/v1alpha1/kafka_conversion.go:42]: Format Go code:
```suggestion
						Enable: source.Spec.KafkaAuthSpec.Net.SASL.Enable,
						User: bindingsv1beta1.SecretValueFromSource{
```

File: kafka/source/pkg/apis/sources/v1alpha1/kafka_conversion.go [hunk 18]
```diff
@@ -0,0 +1,128 @@
...
+							SecretKeyRef: source.Spec.KafkaAuthSpec.Net.SASL.User.SecretKeyRef,
+						},
+						Password: bindingsv1beta1.SecretValueFromSource{
+							SecretKeyRef:source.Spec.KafkaAuthSpec.Net.SASL.Password.SecretKeyRef},
+					},
+					TLS:  bindingsv1beta1.KafkaTLSSpec{
+						Enable: source.Spec.KafkaAuthSpec.Net.TLS.Enable,
+						Cert:   bindingsv1beta1.SecretValueFromSource{
```
Comment (mattmoor) [inline kafka/source/pkg/apis/sources/v1alpha1/kafka_conversion.go:50]: Format Go code:
```suggestion
						Cert: bindingsv1beta1.SecretValueFromSource{
```

File: kafka/source/pkg/apis/sources/v1alpha1/kafka_conversion.go [hunk 19]
```diff
@@ -0,0 +1,128 @@
...
+	switch source := obj.(type) {
+	case *v1beta1.KafkaSource:
+		sink.ObjectMeta = source.ObjectMeta
+		sink.Spec = KafkaSourceSpec{
+			KafkaAuthSpec: v1alpha1.KafkaAuthSpec{
+				BootstrapServers: source.Spec.KafkaAuthSpec.BootstrapServers,
+				Net:              v1alpha1.KafkaNetSpec{
+					SASL: v1alpha1.KafkaSASLSpec{
+						Enable:   source.Spec.KafkaAuthSpec.Net.SASL.Enable,
+						User:     v1alpha1.SecretValueFromSource{
```
Comment (mattmoor) [inline kafka/source/pkg/apis/sources/v1alpha1/kafka_conversion.go:97]: Format Go code:
```suggestion
						Enable: source.Spec.KafkaAuthSpec.Net.SASL.Enable,
						User: v1alpha1.SecretValueFromSource{
```

File: kafka/source/pkg/apis/sources/v1alpha1/kafka_conversion.go [hunk 20]
```diff
@@ -0,0 +1,128 @@
...
+							SecretKeyRef: source.Spec.KafkaAuthSpec.Net.SASL.User.SecretKeyRef,
+						},
+						Password: v1alpha1.SecretValueFromSource{
+							SecretKeyRef:source.Spec.KafkaAuthSpec.Net.SASL.Password.SecretKeyRef},
```
Comment (mattmoor) [inline kafka/source/pkg/apis/sources/v1alpha1/kafka_conversion.go:101]: Format Go code:
```suggestion
							SecretKeyRef: source.Spec.KafkaAuthSpec.Net.SASL.Password.SecretKeyRef},
```

File: kafka/source/pkg/apis/sources/v1alpha1/kafka_conversion.go [hunk 21]
```diff
@@ -0,0 +1,128 @@
...
+					},
+					TLS:  v1alpha1.KafkaTLSSpec{
```
Comment (mattmoor) [inline kafka/source/pkg/apis/sources/v1alpha1/kafka_conversion.go:103]: Format Go code:
```suggestion
					TLS: v1alpha1.KafkaTLSSpec{
```

File: kafka/source/pkg/apis/sources/v1alpha1/kafka_conversion.go [hunk 22]
```diff
@@ -0,0 +1,128 @@
...
+						Enable: source.Spec.KafkaAuthSpec.Net.TLS.Enable,
+						Cert:   v1alpha1.SecretValueFromSource{
```
Comment (mattmoor) [inline kafka/source/pkg/apis/sources/v1alpha1/kafka_conversion.go:105]: Format Go code:
```suggestion
						Cert: v1alpha1.SecretValueFromSource{
```

File: kafka/source/pkg/apis/sources/v1alpha1/kafka_conversion.go [hunk 23]
```diff
@@ -0,0 +1,128 @@
...
+							SecretKeyRef: source.Spec.KafkaAuthSpec.Net.TLS.Cert.SecretKeyRef,
+						},
+						Key:    v1alpha1.SecretValueFromSource{
```
Comment (mattmoor) [inline kafka/source/pkg/apis/sources/v1alpha1/kafka_conversion.go:108]: Format Go code:
```suggestion
						Key: v1alpha1.SecretValueFromSource{
```

File: kafka/source/pkg/apis/sources/v1alpha1/kafka_conversion.go [hunk 24]
```diff
@@ -0,0 +1,128 @@
...
+						},
+						CACert: v1alpha1.SecretValueFromSource{
+							SecretKeyRef: source.Spec.KafkaAuthSpec.Net.TLS.CACert.SecretKeyRef,
+						},
+					},
+				},
+			},
+			Topics:             source.Spec.Topics,
+			ConsumerGroup:      source.Spec.ConsumerGroup,
+			Sink:               source.Spec.Sink.DeepCopy(),
```
Comment (mattmoor) [inline kafka/source/pkg/apis/sources/v1alpha1/kafka_conversion.go:119]: Format Go code:
```suggestion
			Topics:        source.Spec.Topics,
			ConsumerGroup: source.Spec.ConsumerGroup,
			Sink:          source.Spec.Sink.DeepCopy(),
```

File: kafka/source/pkg/apis/sources/v1beta1/kafka_conversion.go [hunk 25]
```diff
@@ -0,0 +1,33 @@
...
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1beta1
+
+import (
+	"context"
+	"fmt"
```
Comment (mattmoor) [inline kafka/source/pkg/apis/sources/v1beta1/kafka_conversion.go:21]: Format Go code:
```suggestion
	"fmt"

```

File: kafka/source/pkg/apis/sources/v1beta1/kafka_conversion.go [hunk 26]
```diff
@@ -0,0 +1,33 @@
...
+)
+
+// ConvertTo implements apis.Convertible
+func (source *KafkaSource) ConvertTo(ctx context.Context, sink apis.Convertible) error {
+	return fmt.Errorf("v1beta1 is the highest known version, got: %T", sink)
+}
+
+// ConvertFrom implements apis.Convertible
+func (sink *KafkaSource) ConvertFrom(ctx context.Context, source apis.Convertible) error {
+	return fmt.Errorf("v1beta1 is the highest known version, got: %T", source)
```
Comment (mattmoor) [inline kafka/source/pkg/apis/sources/v1beta1/kafka_conversion.go:32]: Format Go code:
```suggestion
	return fmt.Errorf("v1beta1 is the highest known version, got: %T", source)
}
```

File: kafka/source/pkg/apis/sources/v1beta1/kafka_validation_test.go [hunk 27]
```diff
@@ -0,0 +1,186 @@
...
+	}{
+		"nil orig": {
+			updated: fullSpec,
+			allowed: true,
+		},
+		"Topic changed": {
+			orig: &fullSpec,
+			updated: KafkaSourceSpec{
+				Topics:             []string{"some-other-topic"},
+				SourceSpec:         fullSpec.SourceSpec,
```
Comment (mattmoor) [inline kafka/source/pkg/apis/sources/v1beta1/kafka_validation_test.go:61]: Format Go code:
```suggestion
				Topics:     []string{"some-other-topic"},
				SourceSpec: fullSpec.SourceSpec,
```

File: kafka/source/pkg/apis/sources/v1beta1/kafka_validation_test.go [hunk 28]
```diff
@@ -0,0 +1,186 @@
...
+			},
+			allowed: false,
+		},
+		"Bootstrap servers changed": {
+			orig: &fullSpec,
+			updated: KafkaSourceSpec{
+				KafkaAuthSpec: bindingsv1beta1.KafkaAuthSpec{
+					BootstrapServers: []string{"server1,server2"},
+				},
+				SourceSpec:  fullSpec.SourceSpec,
```
Comment (mattmoor) [inline kafka/source/pkg/apis/sources/v1beta1/kafka_validation_test.go:71]: Format Go code:
```suggestion
				SourceSpec: fullSpec.SourceSpec,
```

File: kafka/source/pkg/apis/sources/v1alpha1/kafka_conversion.go (hunk 15 above)
Comment (mattmoor) [inline kafka/source/pkg/apis/sources/v1alpha1/kafka_conversion.go:21]: Format Go code:
```suggestion
	"fmt"
	"reflect"

```

File: kafka/source/pkg/apis/sources/v1alpha1/kafka_conversion.go (hunk 16 above)
Comment (mattmoor) [inline kafka/source/pkg/apis/sources/v1alpha1/kafka_conversion.go:27]: Format Go code:
```suggestion

```

File: kafka/source/pkg/apis/sources/v1beta1/kafka_conversion.go (hunk 25 above)
Comment (mattmoor) [inline kafka/source/pkg/apis/sources/v1beta1/kafka_conversion.go:21]: Format Go code:
```suggestion
	"fmt"

```

File: kafka/source/pkg/apis/sources/v1beta1/kafka_types.go (hunk 11 above)
Comment (matzew) [reply kafka/source/pkg/apis/sources/v1beta1/kafka_types.go]: not really - thanks, Updated that @n3wscott

File: kafka/source/pkg/apis/sources/v1beta1/kafka_types.go (hunk 10 above)
Comment (matzew) [reply kafka/source/pkg/apis/sources/v1beta1/kafka_types.go]: thanks for this! I should have looked on the other sources first :-)

File: kafka/source/pkg/apis/sources/v1alpha1/kafka_conversion.go (hunk 15 above)
Comment (mattmoor) [inline kafka/source/pkg/apis/sources/v1alpha1/kafka_conversion.go:21]: Format Go code:
```suggestion
	"fmt"
	"reflect"

```

File: kafka/source/pkg/apis/sources/v1alpha1/kafka_conversion.go (hunk 16 above)
Comment (mattmoor) [inline kafka/source/pkg/apis/sources/v1alpha1/kafka_conversion.go:27]: Format Go code:
```suggestion

```

File: kafka/source/pkg/apis/sources/v1beta1/kafka_conversion.go (hunk 25 above)
Comment (mattmoor) [inline kafka/source/pkg/apis/sources/v1beta1/kafka_conversion.go:21]: Format Go code:
```suggestion
	"fmt"

```

File: kafka/source/pkg/apis/sources/v1alpha1/kafka_conversion.go (hunk 15 above)
Comment (mattmoor) [inline kafka/source/pkg/apis/sources/v1alpha1/kafka_conversion.go:21]: Format Go code:
```suggestion
	"fmt"
	"reflect"

```

File: kafka/source/pkg/apis/sources/v1alpha1/kafka_conversion.go (hunk 16 above)
Comment (mattmoor) [inline kafka/source/pkg/apis/sources/v1alpha1/kafka_conversion.go:27]: Format Go code:
```suggestion

```

File: kafka/source/pkg/apis/sources/v1beta1/kafka_conversion.go (hunk 25 above)
Comment (mattmoor) [inline kafka/source/pkg/apis/sources/v1beta1/kafka_conversion.go:21]: Format Go code:
```suggestion
	"fmt"

```

File: kafka/source/pkg/apis/sources/v1alpha1/kafka_conversion.go (hunk 15 above)
Comment (mattmoor) [inline kafka/source/pkg/apis/sources/v1alpha1/kafka_conversion.go:21]: Format Go code:
```suggestion
	"fmt"
	"reflect"

```

File: kafka/source/pkg/apis/sources/v1alpha1/kafka_conversion.go (hunk 16 above)
Comment (mattmoor) [inline kafka/source/pkg/apis/sources/v1alpha1/kafka_conversion.go:27]: Format Go code:
```suggestion

```

File: kafka/source/pkg/apis/sources/v1beta1/kafka_conversion.go (hunk 25 above)
Comment (mattmoor) [inline kafka/source/pkg/apis/sources/v1beta1/kafka_conversion.go:21]: Format Go code:
```suggestion
	"fmt"

```

File: kafka/source/pkg/apis/sources/v1alpha1/kafka_conversion.go (hunk 15 above)
Comment (mattmoor) [inline kafka/source/pkg/apis/sources/v1alpha1/kafka_conversion.go:21]: Format Go code:
```suggestion
	"fmt"
	"reflect"

```

File: kafka/source/pkg/apis/sources/v1alpha1/kafka_conversion.go (hunk 16 above)
Comment (mattmoor) [inline kafka/source/pkg/apis/sources/v1alpha1/kafka_conversion.go:27]: Format Go code:
```suggestion

```

File: kafka/source/pkg/apis/sources/v1beta1/kafka_conversion.go (hunk 25 above)
Comment (mattmoor) [inline kafka/source/pkg/apis/sources/v1beta1/kafka_conversion.go:21]: Format Go code:
```suggestion
	"fmt"

```

**Comment by aliok** [comment]
Created https://github.com/knative/eventing-contrib/issues/1332 for reconciler work

**Pull request merged**